pkgs/development/compilers/elm/packages/node-composition.nix

```diff
@@ -1,4 +1,4 @@
-# This file has been generated by node2nix 1.8.0. Do not edit!
+# This file has been generated by node2nix 1.7.0. Do not edit!
```

Comment:
I'm a bit behind on my machine because I require Haskell's dbus package for my setup to work thus there is this downgrade.

Reply:
node2nix should come from nixpkgs, but out of scope for this issue :)

Reply:
Agree. I originally copied [this script](https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/compilers/elm/packages/generate-node-packages.sh) from some other place but as you can see it uses `nix-shell -i bash -p nodePackages.node2nix` so system's version of `<nixpkgs>`. I'm not sure how can this be fixed at the moment but it seems to me it would be better if this used `node2nix` from repository itself.

Reply:
something like:

```bash

ROOT="$(realpath "$(dirname -- "$(readlink -f -- "${BASH_SOURCE[0]}")")"/../../../../..)"
$(nix-build $ROOT -A nodePacakges.node2nix --no-out-link)/bin/node2nix ...
```